packages/cli/src/commands/hydrogen/upgrade.ts

```diff
@@ -0,0 +1,2931 @@
+        return acc;
+      },
+      {featuresMd: [], breakingChangesMd: []} as {
+        featuresMd: string[];
+        breakingChangesMd: string[];
+      },
+    );
+
+  const fixesMd = cumulativeRelease.fixes
+    .filter((fixes) => fixes.steps)
+    .map(generateStepMd);
+
+  if (!featuresMd.length && !fixesMd.length) {
+    renderInfo({
+      headline: `No upgrade instructions generated`,
+      body: `There are no additional upgrade instructions for this version.`,
+    });
+    return;
+  }
+
+  const absoluteFrom = getAbsoluteVersion(currentVersion);
+  const absoluteTo = getAbsoluteVersion(selectedRelease.version);
+  filename = `${
+    dryRun ? 'preview-' : ''
+  }upgrade-${absoluteFrom}-to-${absoluteTo}.md`;
+
+  const dotShopifyFolderPath = path.join(appPath, '.shopify/');
+
+  const h1 = `# Hydrogen upgrade guide: ${absoluteFrom} to ${absoluteTo}`;
+
+  let md = `${h1}\n\n----\n`;
+
+  if (breakingChangesMd.length) {
+    md += `\n## Breaking changes\n\n${breakingChangesMd.join('\n')}\n----\n`;
+  }
+
+  if (featuresMd.length) {
+    md += `\n## Features\n\n${featuresMd.join('\n')}\n----\n`;
+  }
+
+  if (fixesMd.length) {
+    md += `\n${featuresMd.length ? '----\n\n' : ''}## Fixes\n\n${fixesMd.join(
+      '\n',
+    )}`;
+  }
+
+  const filePath = path.join(dotShopifyFolderPath, filename);
+
+  try {
+    await isDirectory(dotShopifyFolderPath);
+  } catch (error) {
+    await mkdir(dotShopifyFolderPath);
+  }
+
+  if (!(await fileExists(filePath))) {
+    await touchFile(filePath);
+  } else {
+    const overwriteMdFile = await renderConfirmationPrompt({
+      message: `A previous upgrade instrunctions file already exists for this version.\nDo you want to overwrite it?`,
+      defaultValue: false,
+    });
+
+    if (overwriteMdFile) {
+      await removeFile(`${filePath}.old`);
+    } else {
+      return;
+    }
+    await touchFile(filePath);
+  }
+
+  await writeFile(filePath, md);
+
+  return `.shopify/${filename}`;
+}
+
+/**
+ * Displays a notice to the user if there are new Hydrogen versions available
+ * This function is executed by the `dev` command
+ */
+export async function displayDevUpgradeNotice({
+  targetPath,
+}: {
+  targetPath?: string;
+}) {
+  const appPath = targetPath ? path.resolve(targetPath) : process.cwd();
+  const {currentVersion} = await getHydrogenVersion({appPath});
+
+  const changelog = await fetchTempChangelog();
+
+  if (!changelog?.releases) return;
+
+  const {availableUpgrades, uniqueAvailableUpgrades} = getAvailableUpgrades({
+    releases: changelog.releases,
+    currentVersion,
+  });
+
+  if (!availableUpgrades?.length) {
+    renderSuccess({
+      headline: `You are on the latest Hydrogen version: ${getAbsoluteVersion(
+        currentVersion,
+      )}`,
+    });
+    return;
+  }
+
+  if (!availableUpgrades[0]?.version) {
+    renderError({
+      headline: `Failed to find the latest Hydrogen version`,
+      body: `Please try again later`,
+    });
+    return;
+  }
```

Comment:
I think we should just skip any output during dev if the user is on latest or if we can't find the new versions.
Basically, only distract them with information if they need to do something.